stdlib/universe/chande_momentum_oscillator.go

```diff
@@ -0,0 +1,348 @@
+	}
+
+	return &ChandeMomentumOscillatorProcedureSpec{
+		N:       spec.N,
+		Columns: spec.Columns,
+	}, nil
+}
+
+func (s *ChandeMomentumOscillatorProcedureSpec) Kind() plan.ProcedureKind {
+	return ChandeMomentumOscillatorKind
+}
+
+func (s *ChandeMomentumOscillatorProcedureSpec) Copy() plan.ProcedureSpec {
+	ns := new(ChandeMomentumOscillatorProcedureSpec)
+	*ns = *s
+	if s.Columns != nil {
+		ns.Columns = make([]string, len(s.Columns))
+		copy(ns.Columns, s.Columns)
+	}
+	return ns
+}
+
+// TriggerSpec implements plan.TriggerAwareProcedureSpec
+func (s *ChandeMomentumOscillatorProcedureSpec) TriggerSpec() plan.TriggerSpec {
+	return plan.NarrowTransformationTriggerSpec{}
+}
+
+func createChandeMomentumOscillatorTransformation(id execute.DatasetID, mode execute.AccumulationMode, spec plan.ProcedureSpec, a execute.Administration) (execute.Transformation, execute.Dataset, error) {
+	s, ok := spec.(*ChandeMomentumOscillatorProcedureSpec)
+	if !ok {
+		return nil, nil, fmt.Errorf("invalid spec type %T", spec)
+	}
+	cache := execute.NewTableBuilderCache(a.Allocator())
+	d := execute.NewDataset(id, mode, cache)
+	t := NewChandeMomentumOscillatorTransformation(d, cache, s)
+	return t, d, nil
+}
+
+type chandeMomentumOscillatorTransformation struct {
+	d     execute.Dataset
+	cache execute.TableBuilderCache
+
+	n       int64
+	columns []string
+}
+
+func NewChandeMomentumOscillatorTransformation(d execute.Dataset, cache execute.TableBuilderCache, spec *ChandeMomentumOscillatorProcedureSpec) *chandeMomentumOscillatorTransformation {
+	return &chandeMomentumOscillatorTransformation{
+		d:       d,
+		cache:   cache,
+		n:       spec.N,
+		columns: spec.Columns,
+	}
+}
+
+func (t *chandeMomentumOscillatorTransformation) RetractTable(id execute.DatasetID, key flux.GroupKey) error {
+	return t.d.RetractTable(key)
+}
+
+func (t *chandeMomentumOscillatorTransformation) Process(id execute.DatasetID, tbl flux.Table) error {
+	builder, created := t.cache.TableBuilder(tbl.Key())
+	if !created {
+		return fmt.Errorf("chande momentum oscillator found duplicate table with key: %v", tbl.Key())
+	}
+	cols := tbl.Cols()
+	doChandeMomentumOscillator := make([]bool, len(cols))
+	for j, c := range cols {
+		found := false
+		for _, label := range t.columns {
+			if c.Label == label {
+				if c.Type != flux.TInt && c.Type != flux.TUInt && c.Type != flux.TFloat {
+					return fmt.Errorf("cannot take chande momentum oscillator of column %s (type %s)", c.Label, c.Type.String())
+				}
+				found = true
+				break
+			}
+		}
+
+		if found {
+			mac := c
+			mac.Type = flux.TFloat
+			_, err := builder.AddCol(mac)
+			if err != nil {
+				return err
+			}
+			doChandeMomentumOscillator[j] = true
+		} else {
+			_, err := builder.AddCol(c)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return tbl.Do(func(cr flux.ColReader) error {
+		if cr.Len() == 0 {
+			return nil
+		}
+
+		for j, c := range cr.Cols() {
+			if !doChandeMomentumOscillator[j] {
+				switch c.Type {
```

Comment:
I think there's an `AppendValues` function that should be better at this instead of including a type switch for copying over values.

Reply:
Is there an example code snippet? I don't know how to use `AppendValue`. Or how to make type `*array._____` into `[]values.Value`.

Reply:
Nevermind, I finally figured it out.